pkgs/tools/security/eyewitness/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv
+, fetchFromGitHub
+, firefox
+, geckodriver
+, xorg
+, python37
+, python37Packages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "eyewitness";
+  version = "v20201021.1";
+
+  src = fetchFromGitHub {
+    owner = "FortyNorthSecurity";
+    repo = "EyeWitness";
+    rev = version;
+    sha256 = "16ayiklgf9d1sq8phsfpg1kjs40y4dpwzgwgx8vy2n32ckqv8f6i";
+  };
+
+  # These _are_ runtime, not buildtime dependencies. EyeWitness functions by
+  # executing firefox in Xvfb so have a somewhat 'heavy' dependency tree.
+  #
+  # After EyeWitness finishes processing, firefox and Xvfb are terminated.
+  propagatedBuildInputs = [
+    firefox
+    geckodriver
+    xorg.xorgserver
+    python37
```

Comment:
I am not sure what the best approach here is. Usually I just pass a single `python` script into these expressions. That one python version is generated via `python3.withPackages (p: [ p.netaddr p.selenium … ] )`.

- We should probably avoid hard coding python version 3.7 if we can. Just using python 3 should be sufficient.
- Why did you add `pip` as dependency? Does it try to install packages during runtime?

Reply:
cc @FRidh on the matter of how we pass python envs these days.

Reply:
Done.

Reply:
This is still using python3.7 explicitly unless really required we should use python3 here.

Reply:
I think this part needs some revisiting.